### PR TITLE
BaSP-TG-59: Tasks shared components and redux

### PR DESCRIPTION
**Pull request by agusting19**
Implemented the following shared components in Tasks section:
- Preloader
- Table
- Add task modal
- Edit task modal
- Delete confirmation modal
- Success / error modal
- Buttons
- Inputs and select

Everything should be working fine and showing the correct forms and modals.
Implemented redux to handle the states of components and thunks for fetch methods.

### CODE REVIEW

File: src/Components/Tasks/Form/Form.jsx [hunk 1]
```diff
@@ -15,88 +24,85 @@ const Form = () => {
...
-            'https://alexis-trackgenix-app-git-feature-tg-30tasks-agustin-basp-m2022.vercel.app/tasks'
-          );
-        }
-      });
+  const onSubmit = () => {
+    closeModalForm();
+    dispatch(addTask(userInput, setMessage));
+    setShowMessageModal(true);
+    console.log(`message: ${message}`);
+    console.log(`setShowMessage: ${setShowMessageModal}`);
```
Comment (Javier-Cervera) [inline src/Components/Tasks/Form/Form.jsx:32]: console.log

File: src/Components/Tasks/Edit/Edit.jsx [hunk 2]
```diff
@@ -1,119 +1,98 @@
...
-            'https://alexis-trackgenix-app-git-feature-tg-30tasks-agustin-basp-m2022.vercel.app/tasks'
-          );
-        }
-      });
+  const onSubmit = () => {
+    closeModalForm();
+    dispatch(editTask(userInput, task._id, (Response) => setMessage(Response)));
+    setShowMessageModal(true);
+    console.log(`message: ${message}`);
+    console.log(`setShowMessage: ${setShowMessageModal}`);
```
Comment (Javier-Cervera) [inline src/Components/Tasks/Edit/Edit.jsx:32]: console.log

File: src/redux/tasks/thunks.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    return fetch(`${process.env.REACT_APP_API_URL}/api/tasks`, {
+      method: 'POST',
+      headers: {
+        'Content-type': 'application/json'
+      },
+      body: JSON.stringify(task)
+    })
+      .then((response) => response.json())
+      .then((response) => {
+        dispatch(addTaskSucces(response.data));
```
Comment (Javier-Cervera) [inline src/redux/tasks/thunks.js:42]: If we create a Task that doesn't pass validations, the page breaks.
In that case we could use a conditional so the dispatch isn't used
```
        if (response.error === false) {
          dispatch(addTaskSucces(response.data));
        }
```

File: src/redux/tasks/actions.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+export const getTasksError = (error) => ({
+  type: GET_TASKS_ERROR,
+  payload: error
+});
+
+export const addTaskPending = () => ({
+  type: ADD_TASK_PENDING
+});
+
+export const addTaskSucces = (data) => ({
```
Comment (Javier-Cervera) [inline src/redux/tasks/actions.js:34]: For consistency it should be written as "addTaskSuccess".
The same for all the rest

File: src/Components/Tasks/index.js [hunk 5]
```diff
@@ -1,50 +1,120 @@
...
+        isOpen={showConfirmModal}
+        handleClose={closeModal}
+        confirmDelete={handleConfirm}
+        title="Delete Task"
+        message="¿Are you sure you want to delete the task?"
+      />
+    );
+  }
+
+  return isLoading ? (
```
Comment (Javier-Cervera) [inline src/Components/Tasks/index.js:94]: The page reloading with a modal open could be stopped using more conditions
```
 return isLoading &&
    !showModalFormEdit &&
    !showModalFormAdd &&
    !showMessageModal &&
    !showConfirmModal ?
```

**Review by iaracriscenti — APPROVED** [review]
Clean code and working good. Nice job!

**Review by NicolasFontana — APPROVED** [review]
Everything looks nice!

**Comment by ArianaMaldonado** [comment]
Tested locally on: https://alexis-trackgenix-app-git-feature-tg-59-tasks-404db7-basp-m2022.vercel.app/

**Test Result: Failed ⚠️** 

Bugs found:
1. Delete modal needs more space for "Cancel" and "Delete" buttons. Also the "X" button/icon should be like the others: white and with no background as the example in the second picture.
Screenshot from 2022-06-17 20-58-21

Screenshot from 2022-06-17 21-02-27

2. When adding a task with a longer description, the size of the table is changed. This shouldn't happen. _Maybe we can use a max-width?_

**Original table:**
Screenshot from 2022-06-17 21-01-29

**Modified table:**
Screenshot from 2022-06-17 21-08-55

**Review by ArianaMaldonado — CHANGES_REQUESTED** [review]
Changes requested. Please, refer to https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/24#issuecomment-1159313105

**Review by ArianaMaldonado — APPROVED** [review]
**Test Result: Approved.**
_Note: Changes requested to be fixed in "BaSP-TG-71: Fix shared components"._

**Pull request merged**